packages/twenty-server/src/engine/core-modules/billing/commands/billing-sync-plans-data.command.ts

```diff
@@ -0,0 +1,132 @@
+import { StripeService } from 'src/engine/core-modules/billing/stripe/stripe.service';
+import { isStripeValidProductMetadata } from 'src/engine/core-modules/billing/utils/is-stripe-valid-product-metadata.util';
+import { transformStripeMeterDataToMeterRepositoryData } from 'src/engine/core-modules/billing/utils/transform-stripe-meter-data-to-meter-repository-data.util';
+import { transformStripePriceDataToPriceRepositoryData } from 'src/engine/core-modules/billing/utils/transform-stripe-price-data-to-price-repository-data.util';
+import { transformStripeProductDataToProductRepositoryData } from 'src/engine/core-modules/billing/utils/transform-stripe-product-data-to-product-repository-data.util';
+@Command({
+  name: 'billing:sync-plans-data',
+  description:
+    'Fetches from stripe the plans data (meter, product and price) and upserts it into the database',
+})
+export class BillingSyncPlansDataCommand extends BaseCommandRunner {
+  constructor(
+    @InjectRepository(BillingPrice, 'core')
+    private readonly billingPriceRepository: Repository<BillingPrice>,
+    @InjectRepository(BillingProduct, 'core')
+    private readonly billingProductRepository: Repository<BillingProduct>,
+    @InjectRepository(BillingMeter, 'core')
+    private readonly billingMeterRepository: Repository<BillingMeter>,
+    private readonly stripeService: StripeService,
+  ) {
+    super();
+  }
+
+  private async upsertMetersRepositoryData(
+    meters: Stripe.Billing.Meter[],
+    options: BaseCommandOptions,
+  ) {
+    meters.map(async (meter) => {
+      try {
+        if (!options.dryRun) {
+          await this.billingMeterRepository.upsert(
+            transformStripeMeterDataToMeterRepositoryData(meter),
+            {
+              conflictPaths: ['stripeMeterId'],
+            },
+          );
+        }
+        this.logger.log(`Upserted meter: ${meter.id}`);
+      } catch (error) {
+        this.logger.error(`Error upserting meter ${meter.id}: ${error}`);
+      }
+    });
+  }
+
+  private async upsertProductRepositoryData(
+    product: Stripe.Product,
+    options: BaseCommandOptions,
+  ) {
+    try {
+      if (!options.dryRun) {
+        await this.billingProductRepository.upsert(
+          transformStripeProductDataToProductRepositoryData(product),
+          {
+            conflictPaths: ['stripeProductId'],
+          },
+        );
+      }
+      this.logger.log(`Upserted product: ${product.id}`);
+    } catch (error) {
+      this.logger.error(`Error upserting product ${product.id}: ${error}`);
+    }
+  }
+
+  override async executeBaseCommand(
+    passedParams: string[],
+    options: BaseCommandOptions,
+  ): Promise<void> {
+    const billingMeters = await this.stripeService.getAllMeters();
+
+    await this.upsertMetersRepositoryData(billingMeters, options);
+
+    const billingProducts = await this.stripeService.getAllProducts();
+
+    const billingProductsIds = await Promise.all(
```

Comment:
Do we expect a lot of records in that one? If so, Promise.all could benefit from using batching here

Reply:
I estimate that we will have between 10 and 100  in the course of Twenty's life (realistically we will have approx 15). Is that considered enough records to consider the use of a batch?